src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int MAXIMUM_STORAGE_SIZE = 10;
+    private int size = 0;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAXIMUM_STORAGE_SIZE];
+        values = (V[]) new Object[MAXIMUM_STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
+            for (int i = 0; i < size; i++) {
+                if ((keys[i] == null && keys[i] == key)
+                        || (keys[i] != null && keys[i].equals(key))) {
+                    values[i] = value;
+                }
+            }
+        } else if (size <= keys.length && size <= values.length) {
```

Comment:
Its enough to check length only for keys or value
```suggestion
        } else if (size <= keys.length) {
```